packages/affiliates/libs/contracts.js

```diff
@@ -129,6 +131,22 @@ function Emp({ abi = getAbi("ExpiringMultiParty"), web3 } = {}) {
   return { tokenCurrency, collateralCurrency, collateralInfo, tokenInfo, info };
 }
 
+// web3s fromWei does not allow you to specify numeric decimal conversion, it has to be mapped to a name
+function fromWei(amount, decimals = 18) {
+  const denominator = toBN("10").pow(toBN(decimals.toString()));
+  return toBN(amount.toString()).div(denominator);
+}
+// calculate a value given token amount, price and decimals
+// amount: amount of token in wei
+// price: price as a float
+// decimals: decimal value for the token
+// returns value in wei 18 decimals
+function calculateValue(amount, price, decimals) {
+  price = toBN(toWei(price.toString()));
+  amount = toBN(amount);
+  return fromWei(amount.mul(price), decimals);
+}
+
```

Comment:
a couple utility functions added. web3s `fromWei` is really wierd and does not work the same way as ethers, it requires you pass in a string like `wei` or `gwei` which is really useless, so i made a utility